Client/src/Unified/Factory/SessionServiceFactory.cs

```diff
@@ -44,16 +43,15 @@ public class SessionServiceFactory
   ///   The ctor with IConfiguration and optional TaskOptions
   /// </summary>
   /// <param name="loggerFactory">The factory to create the logger for clientService</param>
-  public SessionServiceFactory([CanBeNull] ILoggerFactory loggerFactory = null)
+  public SessionServiceFactory(ILoggerFactory? loggerFactory = null)
   {
-    LoggerFactory = loggerFactory;
-    Logger        = loggerFactory?.CreateLogger<SessionServiceFactory>();
+    LoggerFactory = loggerFactory ?? new NullLoggerFactory();
+    Logger        = LoggerFactory.CreateLogger<SessionServiceFactory>();
   }
 
-  [CanBeNull]
   private ILogger<SessionServiceFactory> Logger { get; }
 
-  private ChannelPool GrpcPool { get; set; }
+  private ChannelPool? GrpcPool { get; set; }
```

Comment:
Check whether this is still required and how it is connected to the pool contained in parent class.